### PR TITLE
Publish manifest for orchestrated build

### PR DESCRIPTION
**Pull request by jaredpar**
Changes our blob publish to include the manifest file. Needed by more
recent versions of product build.

Also decided to just link our two msbuild files together via MSBuild
task vs. a powershell script. The point of microbuild.ps1 is to test as
much as possible in our jenkins build. There is no way to actually test
blob publishing in Jenkins hence this indirection was just causing
problems, not helping anything.

### CODE REVIEW

**Comment by jaredpar** [comment]
Keeping this as WIP until I can get a couple of official builds completed to verify te changes.

File: src/Tools/MicroBuild/PublishBlobAssets.proj [hunk 1]
```diff
@@ -18,10 +17,16 @@
...
+                    AccountKey="$(PB_PublishBlobFeedKey)"
                     ItemsToPush="@(ItemsToPush)"
-                    Overwrite="$(PublishOverwrite)" />
+                    Overwrite="$(PublishOverwrite)"
+                    ManifestBranch="$(BUILD_SOURCEBRANCH)"
+                    ManifestBuildId="$(BUILD_BUILDNUMBER)"
+                    ManifestCommit="$(CommitHashDisplay)"
+                    ManifestName="roslyn"
+                    SkipCreateManifest="false" />
+                    />
```
Comment (dagood) [inline src/Tools/MicroBuild/PublishBlobAssets.proj]: Extra `/>`

File: src/Tools/MicroBuild/PublishBlobAssets.proj [hunk 2]
```diff
@@ -18,10 +17,16 @@
...
+    <PushToBlobFeed ExpectedFeedUrl="$(PB_PublishBlobFeedUrl)"
+                    AccountKey="$(PB_PublishBlobFeedKey)"
                     ItemsToPush="@(ItemsToPush)"
-                    Overwrite="$(PublishOverwrite)" />
+                    Overwrite="$(PublishOverwrite)"
+                    ManifestBranch="$(BUILD_SOURCEBRANCH)"
+                    ManifestBuildId="$(BUILD_BUILDNUMBER)"
+                    ManifestCommit="$(CommitHashDisplay)"
+                    ManifestName="roslyn"
+                    SkipCreateManifest="false" />
```
Comment (dagood) [inline src/Tools/MicroBuild/PublishBlobAssets.proj]: `SkipCreateManifest` is optional and defaults to `false`, not needed. (But doesn't hurt anything of course.)

File: src/Tools/MicroBuild/PublishBlobAssets.proj (hunk 1 above)
Comment (jaredpar) [reply src/Tools/MicroBuild/PublishBlobAssets.proj]: I will not code before finishing coffee
I will not code before finishing coffee
I will not code before finishing coffee

**Comment by jaredpar** [comment]
CC @dotnet/roslyn-infrastructure for review

**Comment by mmitche** [comment]
@jaredpar Is this the right branch?  The branch listed for roslyn is dev15.7x-vs-deps.

**Comment by jaredpar** [comment]
@mmitche yes and yes. This branch auto-merges into dev15.7.x-vs-deps. The vs-deps branch is only editted directly if the change depends on unreleased versions of VS binaries.

**Comment by mmitche** [comment]
@jaredpar Not seeing that auto-merge.  When should it happen?

**Comment by jaredpar** [comment]
It should have happened at most 24 hours after merge. 

@jasonmalinowski are the merges stuck?

**Comment by jasonmalinowski** [comment]
@agocke is managing them this sprint. #24509 would be the PR, looks like it's tied up on some merge conflicts.

**Comment by agocke** [comment]
Yup, I'm working through the PRs right now, so it should be merged shortly. FYI I do them every weekday morning, so if you merge something Friday you won't see it until Monday.

**Pull request merged**